### DISCUSSION TITLE
LittleFS

### DISCUSSION THREAD

**Discussion by MichaelPNolan** [marcel-licence/esp32_drum_computer#6] [General]
So far I can't get a single file or directory read - i bought a new card. I formatted it many ways I used this 
https://github.com/bluscape/LittleFS-Explorer-for-Windows
tool to format it after a while of figuring out I was getting nowhere.

So - what is going on, is this project going to us the SD card or partitioned memory? 
Now that I'm looking at this https://www.youtube.com/watch?v=4r6YZlLfKfw - i have a bit more clue what seems to be going on. Never really thought much about this data directory or this partitioning until today.

Regards
Michael

**Comment by MichaelPNolan** [comment]
This is where I am - I can upload the files with the tools. The files aren't opening though ... do I need a different library I wonder.
```
Loading data
Firmware started successfully
Connect to AC101 codec... OK
Listing directory: /
  FILE: 440.wav	SIZE: 22096
  FILE: 880.wav	SIZE: 22096
  FILE: white_noise.wav	SIZE: 22096
---
ListSamples:
s[0]: 440.wav
error openening file!
s[1]: 880.wav
error openening file!
s[2]: white_noise.wav
error openening file!
Setup Serial2 with 31250 baud with rx: 21 only
Setup MidiPort2 using Serial2
rev: 15329, 15329
Reverb is ready!
ESP.getFreeHeap() 210999
ESP.getMinFreeHeap() 208627
ESP.getHeapSize() 277379
ESP.getMaxAllocHeap() 65524
Total heap: 277379
Free heap: 210999
Total PSRAM: 4192127
Free PSRAM: 4130811
```

**Reply by marcel-licence** [reply]
The error occurs in case the following line fails:
`File f = LittleFS.open(samplePlayer[i].filename);`

You should try the LittleFS example. I am not sure if something else is broken. I started testing and using ESP32 arduino board version 2.0.2. I pushed some changes into the repo some days ago

**Reply by marcel-licence** [reply]
I am surprised that you are talking about a SD card. LittleFS is used on the ESP32 memory. The executed code and some additional data are stored on flash and separated in two parts. That's why the upload tool is required. The drum computer does not use the SD card (yet).

**Comment by MichaelPNolan** [comment]
Yes, I came to this realization, having no experience with storing data in
the LittleFS. Now, however, i still can't get files to read in so that is
still a problem I have to find.
Getting the tools installed to write the partition to memory - i got
through that. Yes, I was fully lost assuming you used an SD card for
samples.
My drum samples will take a lot of memory - i have to figure out the right
bitrate to make it realistic.

Regards
Michael

**Comment by tiretrak** [comment]
i have the same problem on esp32 audio kit  A1S with 8388 v2.2

ets Jun  8 2016 00:22:57

rst:0x1 (POWERON_RESET),boot:0x1f (SPI_FAST_FLASH_BOOT)
configsip: 0, SPIWP:0xee
clk_drv:0x00,q_drv:0x00,d_drv:0x00,cs0_drv:0x00,hd_drv:0x00,wp_drv:0x00
mode:DIO, clock div:1
load:0x3fff0030,len:1344
load:0x40078000,len:13516
load:0x40080400,len:3604
entry 0x400805f0

Loading data
Firmware started successfully
Connect to ES8388 codec... 0x00: 0x05
0x01: 0x40
AdcCh1!
MixChAMPL!
ES8388 setup finished!
Reg 0x00 = 0x05
Reg 0x01 = 0x40
Reg 0x02 = 0x00
Reg 0x03 = 0x00
Reg 0x04 = 0x3c
Reg 0x05 = 0x00
Reg 0x06 = 0x00
Reg 0x07 = 0x7c
Reg 0x08 = 0x00
Reg 0x09 = 0x88
Reg 0x0a = 0x50
Reg 0x0b = 0x02
Reg 0x0c = 0x0c
Reg 0x0d = 0x02
Reg 0x0e = 0x30
Reg 0x0f = 0x20
Reg 0x10 = 0x00
Reg 0x11 = 0x00
Reg 0x12 = 0x16
Reg 0x13 = 0xb0
Reg 0x14 = 0x32
Reg 0x15 = 0x06
Reg 0x16 = 0x00
Reg 0x17 = 0x18
Reg 0x18 = 0x02
Reg 0x19 = 0x22
Reg 0x1a = 0x00
Reg 0x1b = 0x02
Reg 0x1c = 0x08
Reg 0x1d = 0x00
Reg 0x1e = 0x1f
Reg 0x1f = 0xf7
Reg 0x20 = 0xfd
Reg 0x21 = 0xff
Reg 0x22 = 0x1f
Reg 0x23 = 0xf7
Reg 0x24 = 0xfd
Reg 0x25 = 0xff
Reg 0x26 = 0x1b
Reg 0x27 = 0xb8
Reg 0x28 = 0x28
Reg 0x29 = 0x28
Reg 0x2a = 0xb8
Reg 0x2b = 0x80
Reg 0x2c = 0x00
Reg 0x2d = 0x00
Reg 0x2e = 0x1e
Reg 0x2f = 0x1e
Reg 0x30 = 0x1e
Reg 0x31 = 0x1e
Reg 0x32 = 0x00
Reg 0x33 = 0xaa
Reg 0x34 = 0xaa
Listing directory: /
  FILE: 440.wav	SIZE: 22096
  FILE: 880.wav	SIZE: 22096
  FILE: white_noise.wav	SIZE: 22096
---
ListSamples:
s[0]: 440.wav
error openening file!
s[1]: 880.wav
error openening file!
s[2]: white_noise.wav
error openening file!
Setup Serial2 with 31250 baud with rx: 21 only
Setup MidiPort2 using Serial2
rev: 15329, 15329
Reverb is ready!
ESP.getFreeHeap() 207920
ESP.getMinFreeHeap() 207108
ESP.getHeapSize() 241520
ESP.getMaxAllocHeap() 110580
Total heap: 241520
Free heap: 207920
Total PSRAM: 4192103
Free PSRAM: 4130267

**Reply by marcel-licence** [reply]
The error is generated by 
`File f = LittleFS.open(samplePlayer[i].filename);` in line 200 in sampler.ino.
It might require some further analysis to figure out what went wrong here.

**Reply by marcel-licence** [reply]
I've updated the project. It seems that the '/' was missing in front of the filename. The open function was not able to find the requested file without selecting a path.

**Comment by MichaelPNolan** [comment]
Ok - cool, i'd like to give it another try! Thanks
